### ISSUE TITLE
Refinement of `unneeded_override` Rule to Properly Handle Swift Attributes

### ISSUE THREAD

**Issue by KosFM** [realm/SwiftLint#5753]
### New Issue Checklist

- [X] I've Updated SwiftLint to the latest version.
- [X] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Feature or Enhancement Proposal

The `unneeded_override` rule in SwiftLint currently triggers a warning for methods that are overridden solely to add Swift attributes, such as `@objc`, `@IBAction`, and similar. Consider the following example:
```swift
class BaseClass {
    func foo() {}
}

class SubClass: BaseClass {
    <<Any Swift Function Attribute (@objc, @IBAction etc.)>> override func foo() { 
        super.foo()
    }
}
```
In this case, SwiftLint issues a warning on the overridden foo() method, indicating that the override is unnecessary. However, the override is intentionally used to apply a specific attribute, which is a legitimate use case.

This raises a discussion point: while attributes on overridden methods may be necessary, the current rule does not appropriately account for this scenario. The rule should be refined to differentiate between purely redundant overrides and those that serve the purpose of attribute application. 

### Environment
* SwiftLint version: 0.56.1
* Xcode version: Xcode 15.4, Build version 15F31d
* Installation method used: Cocoapods

**Comment by SimplyDanny** [comment]
I agree that the rule should ignore overridden methods with attributes.
